### PR TITLE
[FIX] force np.float64 format for second level matrix dataframe

### LINKED ISSUE #4495: :rotating_light: NIGHTLY DEPENDENCIES TEST: failure on `refs/heads/main`
The run `10297598102` of the workflow testing Nilearn with the nightly build of its dependencies failed on `refs/heads/main`.

You can view the report here: https://github.com/nilearn/nilearn/actions/runs/10297598102

Pinging @nilearn/core

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4495 

Changes proposed in this pull request:

- force np.float64 format to make sure compatibility with pandas 3.0

### CODE REVIEW

File: .github/workflows/nightly_dependencies.yml [hunk 1]
```diff
@@ -19,6 +19,10 @@ on:
 
     workflow_dispatch:
 
+# Force to use color
+env:
+    FORCE_COLOR: true
+
```
Comment (Remi-Gau) [inline .github/workflows/nightly_dependencies.yml:25]: just to allow to have color when checking CI STDOUT

**Comment by Remi-Gau** [comment]
this seems to have fixed it: https://github.com/nilearn/nilearn/actions/runs/10286471996/job/28467204312?pr=4503

File: nilearn/glm/first_level/design_matrix.py [hunk 2]
```diff
@@ -488,10 +488,10 @@ def make_second_level_design_matrix(subjects_label, confounds=None):
         raise ValueError("Design matrix columns do not have unique names")
 
     # float dtype necessary for linalg
-    design_matrix = pd.DataFrame(columns=design_columns, dtype=float)
+    design_matrix = pd.DataFrame(columns=design_columns, dtype=np.float64)
```
Comment (bthirion) [inline nilearn/glm/first_level/design_matrix.py:491]:
```suggestion
    design_matrix = pd.DataFrame(columns=design_columns, dtype="float64")
```

**Review by bthirion — COMMENTED** [review]
LGTM overall.

**Review by bthirion — APPROVED** [review]
LGTM, thx

**Comment by bthirion** [comment]
Have you seen the nightly dependencies test failure ? Could it be related to this PR ?

**Comment by Remi-Gau** [comment]
> Have you seen the nightly dependencies test failure ? Could it be related to this PR ?

yup: the current PR (#4503) will fix this.

should we merge with only one review?

**Comment by bthirion** [comment]
@man-shu can you provide quickly a review ?

**Review by man-shu — APPROVED** [review]
LGTM too!

**Comment by bthirion** [comment]
Merging then.

**Pull request merged**